### PR TITLE
changes for entities, need to update tests

### PR DESCRIPTION
**Pull request by tsuwandy**
Fixes #
#5339 

## Description
- Detect entity model in model path, if entity model exist, try scoring entities.
- Update Orchestrator telemetry info, fix top intent query, return 2nd top intent (@daveta )

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs [hunk 1]
```diff
@@ -77,6 +82,11 @@ public OrchestratorAdaptiveRecognizer(string modelFolder, string snapshotFile, I
             InitializeModel();
         }
 
+        [JsonIgnore]
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+        public bool ScoreEntities { get; set; } = false;
```
Comment (daveta) [inline libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs:87]: Just wondering about usage - seems like InitializeModel() will override this depending if entity folder exists.

**Review by daveta — APPROVED** [review]

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs [hunk 2]
```diff
@@ -215,13 +221,140 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dc, Sc
...
             TrackRecognizerResult(dc, $"{nameof(OrchestratorAdaptiveRecognizer)}Result", FillRecognizerResultTelemetryProperties(recognizerResult, telemetryProperties, dc), telemetryMetrics);
 
+            if (ExternalEntityRecognizer != null)
+            {
+                // Run external recognition
+                var externalResults = await ExternalEntityRecognizer.RecognizeAsync(dc, activity, cancellationToken, telemetryProperties, telemetryMetrics).ConfigureAwait(false);
+                recognizerResult.Entities = externalResults.Entities;
+            }
+
+            TryScoreEntities(text, recognizerResult);
```
Comment (congysu) [inline libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs:235]: Will the ExternalEntityRecognizer score a text for entities?  Seems not, and is that the reason we invoke `TryScoreEntities` later?  Thanks.

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs [hunk 3]
```diff
@@ -215,13 +221,140 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dc, Sc
                 // Return 'None' if no intent matched.
                 recognizerResult.Intents.Add(NoneIntent, new IntentScore() { Score = 1.0 });
             }
-
+
```
Comment (congysu) [inline libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs:224]: nit: Feel free to ignore me.  Thanks.

**Review by congysu — APPROVED** [review]
Thanks.

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs (hunk 1 above)
Comment (tsuwandy) [reply libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs:87]: Yeah, this is for unit testing (MockResolver, no real model/modelfolder)

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs (hunk 2 above)
Comment (tsuwandy) [reply libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs:235]: Yes, there are a few built in external entity recognizers (ie regex, etc).  TryScoreEntities is calling Orchestrator Score() for LabelType.Entity.

**Review by mrivera-ms — APPROVED** [review]

**Pull request merged**